Refactor enum in WebTouchEvent to enum class

jesxilin opened the pull request:
#### 26d7460d3e657de37f47b9d8f587e14ec95c0438
<pre>
Refactor enum in WebTouchEvent to enum class
<a href="https://bugs.webkit.org/show_bug.cgi?id=257223">https://bugs.webkit.org/show_bug.cgi?id=257223</a>
rdar://109736930

Reviewed by Wenson Hsieh.

Refactor enum TouchPointState in WebTouchEvent to be enum classes.
Changed TouchPointState to State to reduce redundancy.

* Source/WebKit/Shared/WebEvent.serialization.in:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::touchEventType):
(WebKit::WebKit2PlatformTouchPoint::WebKit2PlatformTouchPoint):
* Source/WebKit/Shared/WebPlatformTouchPoint.cpp:
(WebKit::WebPlatformTouchPoint::WebPlatformTouchPoint):
(WebKit::WebPlatformTouchPoint::encode const): Deleted.
(WebKit::WebPlatformTouchPoint::decode): Deleted.
* Source/WebKit/Shared/WebTouchEvent.cpp:
(WebKit::WebTouchEvent::allTouchPointsAreReleased const):
* Source/WebKit/Shared/WebTouchEvent.h:
(WebKit::WebPlatformTouchPoint::WebPlatformTouchPoint):
(WebKit::WebPlatformTouchPoint::phase const):
(WebKit::WebPlatformTouchPoint::state const):
(WebKit::WebPlatformTouchPoint::setTouchType):
(WebKit::WebPlatformTouchPoint::touchType const):
(WebKit::WebPlatformTouchPoint::setState):
* Source/WebKit/Shared/ios/NativeWebTouchEventIOS.mm:
(WebKit::convertTouchPhase):
(WebKit::NativeWebTouchEvent::extractWebTouchPoint):
* Source/WebKit/Shared/ios/WebPlatformTouchPointIOS.cpp: Removed.
* Source/WebKit/Shared/libwpe/WebEventFactory.cpp:
(WebKit::stateForTouchPoint):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(appendTouchEvent):
(touchPointStateForEvents):
(webkitWebViewBaseGetTouchPointsForEvent):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::touchEventTrackingType const):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _handleTouchActionsForTouchEvent:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/265370@main">https://commits.webkit.org/265370@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f49461b9179b376d8779869333aec3070c6885ef

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10681 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10902 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11191 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12330 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10240 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13275 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10872 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13152 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10841 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11750 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8971 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12732 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9044 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16891 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10114 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9781 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13034 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10252 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/8344 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9412 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2566 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13684 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10114 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/Shared/WebTouchEvent.h:
```diff
@@ -61,8 +61,8 @@ class WebPlatformTouchPoint {
     WebCore::IntPoint location() const { return m_location; }
-    TouchPointState phase() const { return static_cast<TouchPointState>(m_phase); }
-    TouchPointState state() const { return phase(); }
+    State phase() const { return static_cast<State>(m_phase); }
```
this static_cast is now unnecessary I think

rr-codes commented on Source/WebKit/Shared/WebTouchEvent.h:
```diff
@@ -148,31 +148,31 @@ class WebTouchEvent : public WebEvent {
-        TouchMoved,
-        TouchStationary,
-        TouchCancelled
+    enum class State {
```
missing ` : uint8_t`

rr-codes commented on Source/WebKit/Shared/WebTouchEvent.h:
```diff
@@ -148,31 +148,31 @@ class WebTouchEvent : public WebEvent {
     
     uint32_t id() const { return m_id; }
-    TouchPointState state() const { return static_cast<TouchPointState>(m_state); }
+    State state() const { return static_cast<State>(m_state); }
```
ditto

rr-codes commented on Source/WebKit/Shared/WebTouchEvent.h:
```diff
@@ -148,31 +148,31 @@ class WebTouchEvent : public WebEvent {
     float force() const { return m_force; }
 
-    void setState(TouchPointState state) { m_state = state; }
+    void setState(State state) { m_state = state; }
```
m_state needs to become `State` instead of `uint32_t`

cdumez commented on Source/WebKit/Shared/WebTouchEvent.h:
```diff
@@ -38,21 +38,21 @@ namespace WebKit {
     };
 
-    enum class TouchType {
+    enum class TouchType : uint8_t {
```
Could use `: bool`

jesxilin replied:
Tried changing to bool but did not work :( it was giving an error in the serialization code, presumably bc of this check `assert underlying_type != 'bool'`

pxlcoder commented on Source/WebKit/Shared/WebTouchEvent.h:
```diff
@@ -87,15 +87,15 @@ class WebPlatformTouchPoint {
     double m_altitudeAngle { 0 };
     double m_azimuthAngle { 0 };
-    uint32_t m_touchType { static_cast<uint32_t>(TouchType::Direct) };
+    uint8_t m_touchType { TouchType::Direct };
```
This should be `TouchType` rather than `uint8_t`.

whsieh replied:
Is there a reason this can't be `TouchType m_touchType`?

whsieh reviewed (approved):
(One minor question, LGTM otherwise)

aprotyas commented on Source/WebKit/Shared/WebEvent.serialization.in:
```diff
@@ -107,6 +110,40 @@ class WebKit::WebTouchEvent : WebKit::WebEvent {
+    WebKit::WebPlatformTouchPoint::State state()
+    const WebCore::IntPoint& screenPosition()
+    const WebCore::IntPoint& position()
+    const WebCore::IntSize& radius()
```
Do these have to be const refs? I don't see precedence in other `serialization.in` files

jesxilin committed:
Refactor enum in WebTouchEvent to enum class
https://bugs.webkit.org/show_bug.cgi?id=257223
rdar://109736930

Reviewed by Wenson Hsieh.

Refactor enum TouchPointState in WebTouchEvent to be enum classes.
Changed TouchPointState to State to reduce redundancy.

* Source/WebKit/Shared/WebEvent.serialization.in:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::touchEventType):
(WebKit::WebKit2PlatformTouchPoint::WebKit2PlatformTouchPoint):
* Source/WebKit/Shared/WebPlatformTouchPoint.cpp:
(WebKit::WebPlatformTouchPoint::WebPlatformTouchPoint):
(WebKit::WebPlatformTouchPoint::encode const): Deleted.
(WebKit::WebPlatformTouchPoint::decode): Deleted.
* Source/WebKit/Shared/WebTouchEvent.cpp:
(WebKit::WebTouchEvent::allTouchPointsAreReleased const):
* Source/WebKit/Shared/WebTouchEvent.h:
(WebKit::WebPlatformTouchPoint::WebPlatformTouchPoint):
(WebKit::WebPlatformTouchPoint::phase const):
(WebKit::WebPlatformTouchPoint::state const):
(WebKit::WebPlatformTouchPoint::setTouchType):
(WebKit::WebPlatformTouchPoint::touchType const):
(WebKit::WebPlatformTouchPoint::setState):
* Source/WebKit/Shared/ios/NativeWebTouchEventIOS.mm:
(WebKit::convertTouchPhase):
(WebKit::NativeWebTouchEvent::extractWebTouchPoint):
* Source/WebKit/Shared/ios/WebPlatformTouchPointIOS.cpp: Removed.
* Source/WebKit/Shared/libwpe/WebEventFactory.cpp:
(WebKit::stateForTouchPoint):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(appendTouchEvent):
(touchPointStateForEvents):
(webkitWebViewBaseGetTouchPointsForEvent):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::touchEventTrackingType const):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _handleTouchActionsForTouchEvent:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/265370@main
